src/main/app/controller/courts/CasesHeardController.ts

```diff
@@ -0,0 +1,65 @@
+import autobind from 'autobind-decorator';
+import {AuthedRequest} from '../../../types/AuthedRequest';
+import {Response} from 'express';
+import {AreaOfLaw} from '../../../types/AreaOfLaw';
+import {Error} from '../../../types/Error';
+import {CasesHeardPageData} from '../../../types/CasesHeardPageData';
+import {CSRF} from '../../../modules/csrf';
+import {AxiosError} from 'axios';
+
+@autobind
+export class CasesHeardController {
+
+  getAreasOfLawErrorMsg = 'A problem occurred when retrieving the areas of law. ';
+  getCourtAreasOfLawErrorMsg = 'A problem occurred when retrieving the court areas of law. ';
+  putCourtAreasOfLawErrorMsg = 'A problem occurred when updating the court areas of law. ';
+
+  public async get(
+    req: AuthedRequest,
+    res: Response,
+    error = '',
+    updated = false,
+    allAreasOfLaw: AreaOfLaw[] = null,
+    courtAreasOfLaw: AreaOfLaw[]): Promise<void> {
+    const slug: string = req.params.slug as string;
+    if (!allAreasOfLaw) {
+      await req.scope.cradle.api.getAllAreasOfLaw()
+        .then((value: AreaOfLaw[]) => allAreasOfLaw = value)
+        .catch(() => error += this.getAreasOfLawErrorMsg);
```

Comment:
Something that has been picked up on some other tabs is that if either of the API calls fail, we still display an active save button, which I think will be the case here. I think we should try and avoid having this issue on more tabs. You could do something like setting a boolean when either of the critical API calls fail, passing this through to the view and then using that boolean to determine if we display (or enable) the save button or not.

Reply:
Could this be resolved by moving the button inside the if statement so it only renders if the checkboxes are rendered? 

{% if allAreasOfLaw|length %} 

... 

  <div class="govuk-button-group">
    {{ govukButton({
      text: "Update",
      name: "updateCasesHeard",
      id: "updateCasesHeardSelect"
    }) }}
  </div>
{% endif %}

Or is it better to add a boolean because that deals with either of the API calls failing?

Reply:
There are 2 API calls here - getAllAreasOfLaw and getCourtAreasOfLaw. I think that would only cover one of these failing.